brave/brave-core
# feat: Onboarding redesign

Resolves https://github.com/brave/brave-browser/issues/22591
Fixes: https://github.com/brave/brave-browser/issues/22817

Security review request: https://github.com/brave/security/issues/951

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

Prep:
1. Reset Wallet via wallet settings
2. Open wallet via top-bar icon
3. Wallet Onboarding Welcome Panel should appear
4. Click "Learn more"
5. The Wallet Onboarding Welcome Tab should open

https://user-images.githubusercontent.com/30185185/177829416-ed0e5c30-8e3b-439a-a776-8471be90b8f7.mov


Scenario 1 - Learn More about wallets
1. Click "Learn more about crypto wallet"
2. A new tab should open to "https://brave.com/learn/what-is-crypto-wallet/"

https://user-images.githubusercontent.com/30185185/177829569-04a4c7b4-6310-4432-9431-d5762ceff1c0.mov



Scenario 1a - New Wallet
1. Click "Create a new wallet"
2. Agree to Terms
3. Create a password
4. Click "Got it"
5. Backup recovery phrase (check that reveal, clipboard and file download works)
7. Verify recovery phrase
8. Click Next
9. Onboarding success screen appears
10. Click Done
11. Wallet Portfolio page is opened

https://user-images.githubusercontent.com/30185185/177830113-3495d21d-5626-4548-9cc7-717d4daf218c.mov


Scenario 1b - Restore Wallet from recovery phrase

https://user-images.githubusercontent.com/30185185/177831188-eaf04550-b805-468a-ac6e-ba70f4a18a4b.mov

Scenario 1c - Import Wallet from MetaMask

https://user-images.githubusercontent.com/30185185/177862454-448104bc-2441-418e-991a-5a302bc596ba.mov


Scenario 1d - Import Wallet from Legacy Crypto Wallets

https://user-images.githubusercontent.com/30185185/177862819-35b7e355-25ee-48d3-8e08-f24b3a446619.mov

Scenario 2a - Buy Crypto

https://user-images.githubusercontent.com/30185185/177863360-09044e7d-4d01-4063-a9d1-4b866938f19c.mov

Scenario 2b - Fund Wallet with Crypto

https://user-images.githubusercontent.com/30185185/177863699-f84b140f-eee0-4f21-9528-b2b1ab31449a.mov
